homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -265,12 +275,20 @@ def _handle_none_keypad_serial(keypad_device: dict, bridge_serial: int) -> str:
     return keypad_device["serial"] or f"{bridge_serial}_{keypad_device['device_id']}"
 
 
-def _area_and_name_from_name(device_name: str) -> tuple[str, str]:
-    """Return the area and name from the devices internal name."""
-    if "_" in device_name:
-        area_device_name = device_name.split("_", 1)
-        return area_device_name[0], area_device_name[1]
-    return UNASSIGNED_AREA, device_name
+def _area_name_from_id(areas: dict[str, dict], area_id: str) -> str:
+    """Return the full area name including parent(s)."""
+
+    if area_id is None:
+        return UNASSIGNED_AREA
+
+    if "parent_id" in areas[area_id]:
```

Comment:
Please unpack the dict into a new variable to avoid chained lookups

Reply:
done